packages/contracts/test/registry.ts

```diff
@@ -1,46 +1,58 @@
-import {expect} from 'chai';
-import {ethers} from 'hardhat';
-import {customError} from './test-utils/custom-error-helper';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { customError } from './test-utils/custom-error-helper';
 
-const EVENTS = {
-  NewDAORegistered: 'NewDAORegistered',
-};
+import { DAO, Registry } from '../typechain';
+
+const REGISTER_DAO_ROLE = ethers.utils.id('REGISTER_DAO_ROLE')
 
 describe('Registry', function () {
-  let registry: any;
+  let registry: Registry;
+  let dao: DAO;
   let ownerAddress: string;
 
-  before(async () => {
-    const signers = await ethers.getSigners();
+  beforeEach(async function () {
+    const signers = await ethers.getSigners()
     ownerAddress = await signers[0].getAddress();
-  });
 
-  beforeEach(async function () {
+    const DAO = await ethers.getContractFactory('DAO');
+    dao = await DAO.deploy();
+    await dao.initialize('0x', ownerAddress, ethers.constants.AddressZero);
+
     const Registry = await ethers.getContractFactory('Registry');
     registry = await Registry.deploy();
+    await registry.initialize(dao.address);
```

Comment:
Why are we passing it the address of a DAO?
Shouldn't Registry initialization be something self isolated that received requests from a DAO factory?

Reply:
Every component is associated with a DAO. In this case, it is for testing purposes.
For our use-case, this is address of the managing DAO (the AdminDAO) that @Rekard0 and I decided to create to manage the permission of the overall registration process (see comment above).

Reply:
mmm ok, here I wonder whether we are making things extra complex for ourselves. By having the Registry being a Component we get nice features, but at the same we need to carry additional responsibilities (managing a DAO to change the registry).

Reply:
I don't think that it makes things more complicated. In the simplest (and most centralized) way, we give the permissions for the registry (`UPGRADE_ROLE`, `REGISTER_ROLE`) to a simple wallet. Then the DAO is just the container of the ACL. The good thing is that we remain open for every other form of governance.

One thing to consider is that a component is permanently linked with one DAO. To transfer a component to another DAO, we would need to upgrade the `Component` smart contract.
Sarkawt and I would prefer a component because this reduces code duplication and gives us flexibility and control, and minimizes code in the registry by having the permission management separated from the registry.

Reply:
Makes sense, thank you!

Reply:
Still, can we make sure to change the names of the variables to make them more meaningful? Like "dao" => "owningDao" or similar